src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -261,6 +261,31 @@ private void validateTable(TableModel table, Errors errors, SchemaValidationCont
     }
   }
 
+  // Primary Keys and Foreign Keys cannot be arrays or filerefs or dirrefs
+  private void validateColumnType(Errors errors, ColumnModel columnModel, String keyType) {
+    if (columnModel.isArrayOf()) {
+      errors.rejectValue(
+          "schema",
+          String.format("Invalid%s", keyType),
+          String.format(
+              "%s %s cannot be a column with array type", keyType, columnModel.getName()));
+    }
+    if (columnModel.getDatatype() == TableDataType.DIRREF) {
+      errors.rejectValue(
+          "schema",
+          String.format("Invalid%s", keyType),
+          String.format(
+              "%s %s cannot be a column with dirref type", keyType, columnModel.getName()));
+    }
+    if (columnModel.getDatatype() == TableDataType.FILEREF) {
+      errors.rejectValue(
+          "schema",
+          String.format("Invalid%s", keyType),
+          String.format(
+              "%s %s cannot be a column with fileref type", keyType, columnModel.getName()));
+    }
+  }
+
```

Comment:
Most of this method is code duplication. I think it makes sense to pull the `errors.rejectValue` out into a helper method that can be shared between all three cases. Additionally, the number of `if` statements can be reduced by checking if the column data type exists in a set of invalid key types.